tests/virt_autotest/virsh_virtual_network.pm

```diff
@@ -0,0 +1,335 @@
+# P4 Isolated network
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use set_config_as_glue;
+use version_utils 'is_sle';
+
+
+our $virt_host_bridge = 'br0';
+sub prepare_network {
+    my $config_path = "/etc/sysconfig/network/ifcfg-$virt_host_bridge";
+    if (script_run("[[ -f $config_path ]]") != 0) {
+        assert_script_run("ip link add name $virt_host_bridge type bridge");
+        assert_script_run("ip link set dev $virt_host_bridge up");
+        my $bridge_setup = <<'EOF';
+#!/bin/bash
+ACTIVE_NET=$(ip a|awk -F': ' '/state UP/ {print $2}'|head -n1)
+interface="BOOTPROTO='none'\nSTARTMODE='auto'"
+bridge="BOOTPROTO='dhcp'\nBRIDGE='yes'\nBRIDGE_FORWARDDELAY='0'\nBRIDGE_PORTS='$ACTIVE_NET'\nBRIDGE_STP='off'\nSTARTMODE='auto'"
+> /etc/sysconfig/network/ifcfg-br0.new
+echo -e $bridge >/etc/sysconfig/network/ifcfg-br0
+echo -e $interface >/etc/sysconfig/network/ifcfg-$ACTIVE_NET
+cat /etc/sysconfig/network/ifcfg-br0
+cat /etc/sysconfig/network/ifcfg-$ACTIVE_NET
+systemctl restart network.service;
+EOF
+        script_output($bridge_setup, 900);
+    }
+}
+
+sub restore_standalone {
+    my $standalone_path = "/usr/share/qa/qa_test_virtualization/shared/standalone";
+    if (script_run("[[ -f $standalone_path ]]") == 0) { assert_script_run("source $standalone_path", 60); }
+}
+
+sub destroy_standalone {
+    my $cleanup_path = "/usr/share/qa/qa_test_virtualization/cleanup";
+    if (script_run("[[ -f $cleanup_path ]]") == 0) { assert_script_run("source $cleanup_path", 60); }
+}
+
+sub restart_libvirtd {
+    if (is_sle('>11')) { systemctl 'restart libvirtd'; } else { script_run("service libvirtd restart"); }
+}
+
+sub restore_guests {
+    my $get_vm_hostnames   = "virsh list  --all | grep sles | awk \'{print \$2}\'";
+    my $vm_hostnames       = script_output($get_vm_hostnames, 30, type_command => 0, proceed_on_failure => 0);
+    my @vm_hostnames_array = split(/\n+/, $vm_hostnames);
+    foreach (@vm_hostnames_array)
+    {
+        script_run("virsh destroy $_");
+        script_run("virsh undefine $_");
+        script_run("virsh define $_.xml");
+        upload_logs "$_.xml";
+        script_run("rm -rf $_.xml");
+    }
+}
+
+sub restore_network {
+    my $network_mark = "/etc/sysconfig/network/ifcfg-br0.new";
+    if (script_run("[[ -f $network_mark ]]") == 0) {
+        assert_script_run("ip link set dev $virt_host_bridge down", 60);
+        my $bridge_destroy = <<'EOF';
+#!/bin/bash
+ACTIVE_NET=$(ip a|awk -F': ' '/state UP/ {print $2}'|head -n1)
+interface="BOOTPROTO='dhcp'\nSTARTMODE='auto'"
+echo -e $interface >/etc/sysconfig/network/ifcfg-$ACTIVE_NET
+cat /etc/sysconfig/network/ifcfg-$ACTIVE_NET
+rm -rf /etc/sysconfig/network/ifcfg-br0*
+systemctl restart network.service;
+EOF
+        script_output($bridge_destroy, 900);
+    }
+}
+
+sub destroy_vir_network {
+    my $get_vnet_name   = "virsh net-list --all| tail -2| head -1| awk \'{print \$1}\'";
```

Comment:
Could you please comment such commands to make more clear what is expected from it?

Reply:
Working on here now.

Reply:
Great, had updated code and add the comments to make more clear what is expected.

Reply:
Done